### PR TITLE
Refine arch prefix selection

### PR DESCRIPTION
**Pull request by Neumann-A**
as request in https://github.com/boostorg/cmake/pull/57#issuecomment-2081401809

### CODE REVIEW

File: include/BoostInstall.cmake [hunk 1]
```diff
@@ -162,11 +162,11 @@ function(__boost_install_set_output_name LIB TYPE VERSION)
...
-      if(CMAKE_SYSTEM_PROCESSOR MATCHES "^[Aa][Rr][Mm]"
-           OR CMAKE_SYSTEM_PROCESSOR MATCHES "aarch")
-        set(arch "a")
-      elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^mips")
-        set(arch "m")
+      if(CMAKE_SYSTEM_PROCESSOR MATCHES "(i[3-6]86|amd64)")
+        set(arch "x")
+      else()
+        string(SUBSTRING "${CMAKE_SYSTEM_PROCESSOR}" 0 1 arch)
+        string(TOLOWER "${arch}" arch)
```
Comment (Neumann-A) [inline include/BoostInstall.cmake:169]:
```suggestion
      string(SUBSTRING "${CMAKE_SYSTEM_PROCESSOR}" 0 1 arch)
      string(TOLOWER "${arch}" arch)
      if(CMAKE_SYSTEM_PROCESSOR MATCHES "(i[3-6]86|amd64)")
        set(arch "x")
      endif()
```
Alternative?

Comment (Neumann-A) [reply include/BoostInstall.cmake:169]: Could probably also move       
```
string(SUBSTRING "${CMAKE_SYSTEM_PROCESSOR}" 0 1 arch)
string(TOLOWER "${arch}" arch)
```
out of the complete if block and remove the MSVC if block and merge the selection:
```
    string(SUBSTRING "${CMAKE_SYSTEM_PROCESSOR}" 0 1 arch)
    string(TOLOWER "${arch}" arch)

    if(CMAKE_CXX_COMPILER_ARCHITECTURE_ID MATCHES "^ARM")
      set(arch "a")
    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "(i[3-6]86|[aA][mM][Dd]64)"
        OR CMAKE_CXX_COMPILER_ARCHITECTURE_ID STREQUAL "X86"
        OR CMAKE_CXX_COMPILER_ARCHITECTURE_ID STREQUAL "x64"
      )
      set(arch "x")
    endif()
```

**Comment by pdimov** [comment]
No, CMAKE_SYSTEM_PROCESSOR is incorrect under MSVC, so it's best to keep the two branches separate.

**Comment by pdimov** [comment]
This actually seems to derive `a64` for MinGW, so we don't have it quite right yet.

**Comment by pdimov** [comment]
Ah, CMAKE_SYSTEM_PROCESSOR is "AMD64" there, uppercase. Will fix.

**Pull request merged**